test/codeu/chat/util/TokenizerTest.java

```diff
@@ -0,0 +1,50 @@
+package codeu.chat.util;
+import java.util.*;
+import codeu.chat.util.Tokenizer;
+import org.junit.Test;
+import java.io.IOException;
+import static org.junit.Assert.*;
+public final class TokenizerTest {
+	@Test
+	public void testWithNoQuotes() throws IOException {
+		final Tokenizer tokenizer = new Tokenizer("hello world how are you");
+	 	assertEquals(tokenizer.next(), "hello");
+		assertEquals(tokenizer.next(), "world");
+		assertEquals(tokenizer.next(), "how");
+		assertEquals(tokenizer.next(), "are");
+		assertEquals(tokenizer.next(), "you");
+		assertEquals(tokenizer.next(), null);
+	}
+	@Test
+	public void testWithQuotes() throws IOException {
+		final Tokenizer tokenizer = new Tokenizer("\"hello world\" \"how are you\"");
+		assertEquals(tokenizer.next(), "hello world");
+		assertEquals(tokenizer.next(), "how are you");
+		assertEquals(tokenizer.next(), null);
+	}
+	@Test
+	public void testWithEmptyString() throws IOException {
+		final Tokenizer tokenizer = new Tokenizer("");
+		assertEquals(tokenizer.next(), "");
+		assertEquals(tokenizer.next(), null);
+	}
+	@Test(expected=IOException.class)
+	public void testWithOneQuotation() throws IOException{
+		final Tokenizer tokenizer = new Tokenizer("\"hello world how are you");
+		assertEquals(tokenizer.next(), "\"hello");
+		assertEquals(tokenizer.next(), "world");
+		assertEquals(tokenizer.next(), "how");
+		assertEquals(tokenizer.next(), "are");
+		assertEquals(tokenizer.next(), "you");
+		assertEquals(tokenizer.next(), null);
+	}
+	@Test
+	public void testWithWhiteSpaces() throws IOException {
+		final Tokenizer tokenizer = new Tokenizer("\n \r \t");
```

Comment:
This matches your implementation, but do you really want the whitespace characters as tokens?

If it were me, I would modify Tokenizer to automatically ignore the whitespaces (use `String.trim()` method I suggested last time). It should be fine to do this in a separate pull request as this part is not covered directly by your mini feature.

Also this test is failing, you will still have to fix this.